calico-enterprise/operations/monitor/metrics/recommended-metrics.mdx

```diff
@@ -0,0 +1,514 @@
+
+| Remote cluster connections (out of sync) |                                                              |
+| ---------------------------------------- | ------------------------------------------------------------ |
+| Metric                                   | *count by (instance) (remote_cluster_connection_status != 2)* |
+| Explanation                              | Number of remote cluster connections that are **not** in sync (i.e. resyncing or failing to connect). Each remote cluster will report a *connection_status* value from the following list:<br />- 0 = Not Connected<br />- 1 = Connecting<br />- 2 = In Sync<br />- 3 = Resync in Process<br />- 4 = Config Change Restart Required |
+| Threshold value recommendation           | This value should be 2 if everything is in sync. **Note**: At Typha startup, it is normal to have non-2 values, but it should stabilize at 2 after connections come up. |
+| Threshold breach symptoms                | Typha will not receive updates from the relevant remote clusters. Connected clients will see stale or partial data from remote clusters. |
+| Threshold breach recommendations         | Investigate Typha's logs where remote cluster connectivity events are logged. Ensure the networking between clusters is not experiencing issues. |
+| Priority level                           | Recommended.                                                 |
+
+## Typha client metrics
+
+### Total connections accepted
+
+| Total connections accepted        |                                                              |
+| -------------------------------- | ------------------------------------------------------------ |
+| Metric                           | *typha_connections_accepted*                                 |
+| Explanation                      | Total number of connections accepted over time. This value always increases. |
+| Threshold value recommendation   | A steady increase over time is normal. Counters rising after a Felix or Typha restart is also normal (as clients get rebalanced). Investigate connection counters that rise rapidly with no Felix or Typha restarts. |
+| Threshold breach symptoms        | Counters rising when there are no Felix or Typha restarts, or no action that could cause restarts (an upgrade for example), could indicate unexpected Felix or Typha restarts or issues. |
+| Threshold breach recommendations | Check resource usage on Typha(s) and Kubernetes nodes. Increase resources if needed. |
+| Priority level                   | Optional.                                                    |
+
+### Client connections actively streaming
+
+| Client connections actively streaming |                                                              |
+| ------------------------------------- | ------------------------------------------------------------ |
+| Metric                                | *sum by (instance) (typha_connections_streaming)*            |
```

Comment:
This one is a sum because it has per-cache metrics too.

Reply:
Added note.